### PR TITLE
Support multi-arch image building for couchdb docker image

### PR DESCRIPTION
**Pull request by abdasgupta**
## Overview

This PR is the continuation of https://github.com/apache/couchdb-docker/pull/60 PR. As the author of that PR will not be able to conitnue the work, I am planning to take it to completion. This PR is still dependent on enabling ASF Jenkins CI for couchdb on Power. I will update the PR for Jenkins CI here as soon as I come up with a patch.

## Testing recommendations

`make build` on Power and Intel
`make build ARCH=ppc64le` on Intel and vice-verse. 

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

**Comment by wohali** [comment]
@abdasgupta why do you keep closing and opening this PR?

**Comment by abdasgupta** [comment]
@wohali sorry about that but I was not aware of how to re-run those test cases in travis as some of them were failing though code was okay. So can you tell me if there is any such way to re-run those travis test cases other than closing and re-opening the PRs? Btw, you may review this PR and give me your insight on it. Meanwhile, I will be working on the Jenkins CI you have mentioned else where.

**Comment by wohali** [comment]
Ah, ok. I can re-run test cases by going to the control panel and clicking a button, but only because I'm with the correct Apache project. If you push a change to your branch, it should re-run the changes.

**Comment by abdasgupta** [comment]
yeah, sure @wohali But the problem was that the test cases were failing due to inaccessibility of some remote repo. So no change to PR was needed. It just needed only re-run. Otherwise, I would trigger the re-run through pushing changes in my PR.

**Comment by lag-linaro** [comment]
Please correct me if I'm wrong, but this PR isn't really enabling multi-arch is it?

You're only adding PPC.

Could you add i386 and ARM (32/64) support as well, to truly make it multi-arch please?

File: 2.1.1/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+.PHONY: build
+ARCH ?= amd64
+QEMUVERSION=v2.9.1
+TEMP_DIR:=$(shell mktemp -d)
+BUILD_IMAGE ?= couchdb-$(ARCH)
+
+
+ifeq ($(ARCH),amd64)
+	BASEIMAGE?=debian:jessie
+endif
```
Comment (lag-linaro) [inline 2.1.1/Makefile]: Maybe this should just be an else?

debian:jessie should work for most architectures.

File: 2.1.1/Makefile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+ifeq ($(ARCH),ppc64le)
+	BASEIMAGE?=ppc64le/debian:jessie
+        QEMUARCH=ppc64le
+endif
```
Comment (lag-linaro) [inline 2.1.1/Makefile]: It would be good if you could provide support for other architectures here.

File: 1.7.1/Dockerfile.crossbuild [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+# Install instructions from https://cwiki.apache.org/confluence/display/COUCHDB/Debian
+
+RUN groupadd -g 5984 -r couchdb && useradd -u 5984 -d /opt/couchdb -g couchdb couchdb
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+    ca-certificates \
+    curl \
+    erlang-nox \
+    libicu52 \
+    libmozjs185-1.0 \
```
Comment (lag-linaro) [inline 1.7.1/Dockerfile.crossbuild]: Ideally this should use the 'couch-' version, but that would need to be built and published for all supported arches.

File: 1.7.1/Dockerfile.crossbuild [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+# download dependencies, compile and install couchdb,
+# set correct permissions, expose couchdb to the outside and disable logging to disk
+RUN buildDeps=' \
+    gcc \
+    g++ \
+    erlang-dev \
+    libcurl4-openssl-dev \
+    libicu-dev \
+    libmozjs185-dev \
```
Comment (lag-linaro) [inline 1.7.1/Dockerfile.crossbuild]: As above.

**Comment by wohali** [comment]
I'm closing this PR in favour of the approach outlined in #117.

If you are looking to help with PPC support still, lend a hand in the couchdb-ci and couchdb-pkg repos instead.

**Pull request closed without merging**